src/v/coproc/tests/utils/wasm_event_generator.cc

```diff
@@ -0,0 +1,167 @@
+    const auto val_size = value.size();
+    hkey.append(key.data(), key_size);
+    iobuf hval = bytes_to_iobuf(value);
+    return model::record_header(
+      key_size, std::move(hkey), val_size, std::move(hval));
+}
+
+/// Generates an event with randomized data fields, optionally bounded by event
+/// type
+model::record create_wasm_record(wasm_event e) {
+    iobuf key, value;
+    std::vector<model::record_header> headers;
+    if (e.uuid) {
+        key.append(e.uuid->data(), e.uuid->length());
+    }
+    if (e.script) {
+        value.append(e.script->data(), e.script->length());
+    }
+    if (e.desc) {
+        headers.emplace_back(
+          create_wasm_header("description", std::move(*e.desc)));
+    }
+    if (e.checksum) {
+        headers.emplace_back(
+          create_wasm_header("checksum", std::move(*e.checksum)));
+    }
+    if (e.action) {
+        ss::sstring action_value = (*e.action
+                                    == coproc::wasm_event_action::deploy)
+                                     ? "deploy"
+                                     : "remove";
+        headers.emplace_back(
+          create_wasm_header("action", std::move(action_value)));
+    }
+    /// TODO(rob) a bit of code duplication with
+    /// storage/test/utils/random_batch.cc
+    auto k_z = key.size_bytes();
+    auto v_z = value.size_bytes();
+    auto size = sizeof(model::record_attributes::type) // attributes
+                + vint::vint_size(0)                   // timestamp delta
+                + vint::vint_size(0)                   // offset delta
+                + vint::vint_size(k_z)                 // size of key-len
+                + key.size_bytes()                     // size of key
+                + vint::vint_size(v_z)                 // size of value
+                + value.size_bytes() // size of value (includes lengths)
+                + vint::vint_size(headers.size());
+    for (auto& h : headers) {
+        size += vint::vint_size(h.key_size()) + h.key().size_bytes()
+                + vint::vint_size(h.value_size()) + h.value().size_bytes();
+    }
+    return model::record(
+      size,
+      model::record_attributes(0),
+      0,
+      0,
+      k_z,
+      std::move(key),
+      v_z,
+      std::move(value),
+      std::move(headers));
+}
+
+model::record gen_valid_wasm_event(make_event_type type) {
+    auto uuid = random_generators::gen_alphanum_string(15);
+    if (type == make_event_type::random) {
+        type = (random_generators::get_int(0, 1) == 0)
+                 ? make_event_type::deploy
+                 : make_event_type::remove;
+    }
+    if (type == make_event_type::remove) {
+        return create_wasm_record(wasm_event{
+          .uuid = std::move(uuid),
+          .action = coproc::wasm_event_action::remove});
+    }
+    auto script = random_generators::gen_alphanum_string(15);
+    hash_sha256 h;
+    h.update(script);
+    iobuf buf;
+    auto checksum = h.reset();
```

Comment:
Reviewers, is there a better way to convert an std::array<char, 32> to bytes?

Reply:
You'll have to check the syntax. but:
```cpp
std::array<char, 32> chars;
auto b = bytes(std::begin(chars), std::end(chars));
```

Reply:
Exactly!